packages/rum-core/src/domain/contexts/ciVisibilityContext.spec.ts

```diff
@@ -0,0 +1,64 @@
+import type { Configuration } from '@datadog/browser-core'
+import { Observable } from '@datadog/browser-core'
+import { mockCiVisibilityValues } from '../../../test'
+import type { CiVisibilityContext } from './ciVisibilityContext'
+import { CI_VISIBILITY_TEST_ID_COOKIE_NAME, startCiVisibilityContext } from './ciVisibilityContext'
+
+describe('startCiVisibilityContext', () => {
+  let ciVisibilityContext: CiVisibilityContext
+  let cookieObservable: Observable<{
+    name: string
+    value: string | undefined
+  }>
+  beforeEach(() => {
+    cookieObservable = new Observable()
+  })
+
+  afterEach(() => {
+    ciVisibilityContext.stop()
+  })
+
+  it('sets the ci visibility context defined by Cypress global variables', () => {
+    mockCiVisibilityValues('trace_id_value')
+    ciVisibilityContext = startCiVisibilityContext({} as Configuration, cookieObservable)
+
+    expect(ciVisibilityContext.get()).toEqual({
+      test_execution_id: 'trace_id_value',
+    })
+  })
+
+  it('sets the ci visibility context defined by global cookie', () => {
+    mockCiVisibilityValues('trace_id_value', 'cookies')
+    ciVisibilityContext = startCiVisibilityContext({} as Configuration, cookieObservable)
+
+    cookieObservable.notify({ name: CI_VISIBILITY_TEST_ID_COOKIE_NAME, value: 'trace_id_value' })
```

Comment:
**💬 suggestion**: design this test to get the trace id from `getInitCookie` instead of the observable (in other words, you shouldn't need to `notify` from a cookie change to get it to work)